geode-management/src/main/java/org/apache/geode/management/api/ConfigurationInfo.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.apache.geode.management.api;
+
+import static java.util.stream.Collectors.toList;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import org.apache.geode.management.configuration.AbstractConfiguration;
+import org.apache.geode.management.runtime.RuntimeInfo;
+
+/**
+ * This holds the configuration of a particular entity in the cluster. It includes the id of the
+ * entity and a list of configuration per group on the cluster
+ *
+ * @param <T> the config type
+ * @param <R> the runtimeInfo type
+ */
+public class ConfigurationInfo<T extends AbstractConfiguration<R>, R extends RuntimeInfo> {
```

Comment:
On the topic of naming: given how this class is currently used and the other classes names it seems like this class should include "Result" in it. It is only used as part of the result of a get. I think we should consider also using it in the  result of list but in either case this class is a "result".
The primary thing this class has (compared to other results) is the identity of the thing it describes. So would a good name be "IdentifiableResult"?